packages/syft/src/syft/core/node/common/node_service/oblv/oblv_service.py

```diff
@@ -0,0 +1,769 @@
+
+    Raises:
+        OblvKeyNotFoundError: If no key found.
+        OblvProxyConnectPCRError: If unauthorized deployment code used
+
+    Returns:
+        SuccessResponseMessage: Success message on key pair generation.
+    """
+    _allowed = True
+
+    if _allowed:
+        cli = OblvClient(msg.client.token, msg.client.oblivious_user_id)
+        debug("URL = " + cli.deployment_info(msg.deployment_id).instance.service_url)
+        public_file_name = (
+            os.getenv("OBLV_KEY_PATH", "/app/content")
+            + "/"
+            + os.getenv("OBLV_KEY_NAME", "oblv_key")
+            + "_public.der"
+        )
+        private_file_name = (
+            os.getenv("OBLV_KEY_PATH", "/app/content")
+            + "/"
+            + os.getenv("OBLV_KEY_NAME", "oblv_key")
+            + "_private.der"
+        )
+        depl = cli.deployment_info(msg.deployment_id)
+        if depl.is_deleted:
+            raise OblvEnclaveError(
+                "User cannot connect to this deployment, as it is no longer available."
+            )
+        if depl.is_dev_env:
+            process = subprocess.Popen(  # nosec
+                [
+                    "/usr/local/bin/oblv",
+                    "connect",
+                    "--private-key",
+                    private_file_name,
+                    "--public-key",
+                    public_file_name,
+                    "--url",
+                    depl.instance.service_url,
+                    "--pcr0",
+                    depl.pcr_codes[0],
+                    "--pcr1",
+                    depl.pcr_codes[1],
+                    "--pcr2",
+                    depl.pcr_codes[2],
+                    "--port",
+                    "443",
+                    "--lport",
+                    "3030",
+                    "--disable-pcr-check",
+                ],
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+            )
+        else:
+            process = subprocess.Popen(  # nosec
+                [
+                    "/usr/local/bin/oblv",
+                    "connect",
+                    "--private-key",
+                    private_file_name,
+                    "--public-key",
+                    public_file_name,
+                    "--url",
+                    depl.instance.service_url,
+                    "--pcr0",
+                    depl.pcr_codes[0],
+                    "--pcr1",
+                    depl.pcr_codes[1],
+                    "--pcr2",
+                    depl.pcr_codes[2],
+                    "--port",
+                    "443",
+                    "--lport",
+                    "3030",
+                ],
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+            )
+        while process.poll() is None:
+            d = process.stderr.readline().decode()
+            debug(d)
+            if d.__contains__("Error:  Invalid PCR Values"):
```

Comment:
Same with checking i a string is in "d"

Reply:
Solved by [930bba3](https://github.com/OpenMined/PySyft/pull/7202/commits/930bba34747b97eec3fef12c388e12db60c7e7f4)